docs/proposals/resource-quota.md

```diff
@@ -0,0 +1,404 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Resource Quota
+
+## Background
+
+This document describes proposed enhancements to `ResourceQuota` along with a
+phased set of development milestones.
+
+## Use Cases
+
+* As a cluster administrator, I want the ability to quota:
+ * resources per namespace to prevent resource hogs
+ * resources per kind to control cluster size
+ * compute resources to ensure cluster capacity
+ * compute resources for terminating vs non-terminating workloads to fairly share nodes
+ * compute resources for best-effort vs non-besteffort pods to fairly share nodes
+ * creation rate per kind to prevent abuse
+* As a namespace administrator, I want the ability to quota:
+ * resource consumption based on local understanding of individual workload
+
+## High-level description
+
+This proposal augments the existing API to modify a `ResourceQuota`.
+
+### Filter the set of resources tracked
+
+The set of resources tracked by a `ResourceQuota` can be filtered
+via an optional `LabelSelector` and `FieldSelector`.
+
+A `LabelSelector` matches a resource independent of its kind based on its
+user-defined set of labels.  The label selector supports both equality and
+set-based requirements.  It allows a quota to track resource consumption
+based on local understanding of individual workloads.  For example,
+a quota could match all kinds in a namespace that match the selector
+`tier=cache, environment in (dev,qa)` separately from a quota
+that matched `tier=cache, environment in (prod)`.
```

Comment:
does this mean you can't use a label selector to select items of a specific kind with specific labels? do we select across kinds anywhere else currently?